packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -152,13 +166,42 @@ impl Abigen {
                 use fuels::core::{Detokenize, EnumSelector, ParamType, Tokenizable, Token};
 
                 #code
+                #deployable_trait
 
                 #abi_structs
                 #abi_enums
             }
         })
     }
 
+    fn deployable_trait(&self, name: &Ident) -> TokenStream {
+        if let Some(binary_filepath) = self.binary_filepath.as_ref() {
+            return quote! {
+                #[async_trait]
+                impl Deployable for #name {
+                    async fn deploy_compiled(wallet: LocalWallet, params: TxParameters) -> Self {
```

Comment:
The function should not be called `deploy_compiled` but rather `compile_and_deploy`. Also this function should IMO not be in the ABIgen, it is a function that should be handled on the SDK side (like for instance the `load_sway` or `compile` function) and should not be generated as part of a macro call.

Reply:
> The function should not be called deploy_compiled but rather compile_and_deploy

The SDK isn't responsible for compiling. We are not compiling the Sway code here. We're simply `deploy`ing the `compiled` code, hence `deploy_compiled`.

Reply:
summary of our discussion on Slack:
- The "onchain" workflow (ie loading the contract and deploying it) should not be dependent on the ABIgen workflow
- So we should just compress `load and deploy` into one function, but the contract instantiation `Contract::new()` should be separate. 

@digorithm  this is a decision that creates an extra "line of code" but IMO it's worth it compared to the confusion that could arise from having on-chain deployment being done in the ABIgen, what do you think?